### PR TITLE
qe/prisma-models: delete Model, finish hollowing out of convert() 

### PR DESCRIPTION
**Pull request by tomhoule**
After this commit, prisma_models::convert() is a single allocation.
prisma-models exposes a slightly different API from PSL, on top of it.

This finishes the first phase of https://github.com/prisma/client-planning/issues/265

### CODE REVIEW

File: query-engine/prisma-models/src/convert.rs [hunk 1]
```diff
@@ -1,22 +1,6 @@
-use crate::{builders, InternalDataModel, InternalDataModelRef};
-use once_cell::sync::OnceCell;
+use crate::{InternalDataModel, InternalDataModelRef};
 use std::sync::Arc;
 
 pub fn convert(schema: Arc<psl::ValidatedSchema>) -> InternalDataModelRef {
-    let datamodel = dml::lift(&schema);
```
Comment (pimeys) [inline query-engine/prisma-models/src/convert.rs:6]: can we now get rid of `dml::lift`?

File: query-engine/prisma-models/src/model.rs [hunk 2]
```diff
@@ -1,61 +1,55 @@
...
 
-pub struct Model {
-    pub id: ast::ModelId,
-    pub(crate) name: String,
-    pub(crate) manifestation: Option<String>,
-    pub(crate) fields: OnceCell<Fields>,
-    pub(crate) indexes: OnceCell<Vec<Index>>,
-    pub(crate) primary_identifier: OnceCell<FieldSelection>,
-    pub(crate) dml_model: dml::Model,
+// pub struct Model {
```
Comment (pimeys) [inline query-engine/prisma-models/src/model.rs:8]: que?

**Review by pimeys — APPROVED** [review]
nice

File: query-engine/prisma-models/src/convert.rs (hunk 1 above)
Comment (tomhoule) [reply query-engine/prisma-models/src/convert.rs:6]: PSL tests are left :p

File: query-engine/prisma-models/src/model.rs (hunk 2 above)
Comment (tomhoule) [reply query-engine/prisma-models/src/model.rs:8]: good catch, more code we can delete :fire:

**Pull request merged**